poetry/console/commands/init.py

```diff
@@ -430,20 +430,32 @@ def _parse_requirements(self, requirements: List[str]) -> List[Dict[str, str]]:
 
                     result.append(pair)
                     continue
-            elif (os.path.sep in requirement or "/" in requirement) and cwd.joinpath(
-                requirement
-            ).exists():
-                path = cwd.joinpath(requirement)
+            elif (os.path.sep in requirement or "/" in requirement) and (
```

Comment:
We should take this opportunity to move this bit to pathlib.

Reply:
Most of the code here is using pathlib so I am not sure what you mean

Reply:
Oh I was referring to removing the `(os.path.sep in requirement or "/" in requirement)` check entirely as `Path.exists()` should handle this. 

Also out of interest, pathlib does not seem to handle checks correct for file URIs. We might need to handle that separately (outside of this PR).

```console
>>> pathlib.Path("file://tmp").exists()
False
>>> pathlib.Path("file:/tmp").exists()
False
```

Reply:
It's done like so to check early if the requirement looks like a path before trying to check if it exists

Reply:
Not blocking on this as it's an existing logic. Can deal with this outside the pr.